src/main/java/codeu/controller/ChatServlet.java

```diff
@@ -155,6 +191,7 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
     ArrayList<String> tokenizedMessageContent = new ArrayList();
     String parsedMessageContent = "";
     List<Character> validCharFlags = Arrays.asList('*', '_', '`');
+    String emojiFlag = ":";
```

Comment:
I'm not sure this constant helps. (It took me a bit to realize that line 248 meant that we were looking for a colon.) Maybe try inlining?